analysis/src/Hover.ml

```diff
@@ -98,25 +98,53 @@ let newHover ~full:{file; package} locItem =
          | Const_int64 _ -> "int64"
          | Const_nativeint _ -> "int"))
   | Typed (_, t, locKind) ->
+    let fromConstructorPath ~env path =
+      match References.digConstructor ~env ~package path with
+      | None -> None
+      | Some (_env, {name = {txt}; item = {decl}}) ->
+        if Utils.isUncurriedInternal path then None
+        else Some (decl |> Shared.declToString txt |> codeBlock)
+    in
     let fromType ~docstring typ =
       let typeString = codeBlock (typ |> Shared.typeToString) in
-      let extraTypeInfo =
+      let typeDefinitions =
+        (* Expand definitions of types mentioned in typ.
+           If typ itself is a record or variant, search its body *)
         let env = QueryEnv.fromFile file in
-        match typ |> Shared.digConstructor with
-        | None -> None
-        | Some path -> (
-          match References.digConstructor ~env ~package path with
-          | None -> None
-          | Some (_env, {docstring; name = {txt}; item = {decl}}) ->
-            if Utils.isUncurriedInternal path then None
-            else Some (decl |> Shared.declToString txt, docstring))
-      in
-      let typeString, docstring =
-        match extraTypeInfo with
-        | None -> (typeString, docstring)
-        | Some (extra, extraDocstring) ->
-          (typeString ^ "\n\n" ^ codeBlock extra, extraDocstring)
+        let envToSearch, typesToSearch =
+          match typ |> Shared.digConstructor with
+          | Some path -> (
+            let labelDeclarationsTypes lds =
+              lds |> List.map (fun (ld : Types.label_declaration) -> ld.ld_type)
+            in
+            match References.digConstructor ~env ~package path with
```

Comment:
Unrelated to this PR, but it's a little big confusing that both `References` and `Shared` has a `digConstructor`. Or rather, hard to at a glance understand what `Shared.digConstructor` does (`References.digConstructor` is pretty easy to guess it has to do with references at least 😄 )

Reply:
the shared stuff is leftover that should probably be moved elsewhere